src/libraries/System.IO.FileSystem/tests/FileStream/LockUnlock.cs

```diff
@@ -72,6 +71,22 @@ public void Lock_Unlock_Successful(long fileLength, long position, long length)
             }
         }
 
+        [Theory]
+        [InlineData(FileAccess.Read)]
+        [InlineData(FileAccess.Write)]
+        [InlineData(FileAccess.ReadWrite)]
+        [PlatformSpecific(~TestPlatforms.OSX)]
+        public void Lock_Unlock_Successful_AlternateFileAccess(FileAccess fileAccess)
+        {
+            string path = GetTestFilePath();
+            File.WriteAllBytes(path, new byte[100]);
+
+            using FileStream fs = File.Open(path, FileMode.Open, fileAccess);
+
+            fs.Lock(0, 100);
+            fs.Unlock(0, 100);
+        }
```

Comment:
Can you also add tests validating behavior when opening multiple files at the same time, both when all are read-only and when some are for writing?

Reply:
I think we wouldn't be really able to trigger errors for write locks since they are per-process.
See this test below which already test overlapping locks on Linux:
https://github.com/dotnet/runtime/blob/5b9714a407644e451a9ea1caec7837ef7fbbadb4/src/libraries/System.IO.FileSystem/tests/FileStream/LockUnlock.cs#L155-L156

Reply:
> I think we wouldn't be really able to trigger errors for write locks since they are per-process.

The cited test uses RemoteExecutor to validate with multiple processes.  Why is that not applicable here?

Reply:
> Why is that not applicable here?

It is, didn't notice that class, sorry. 

I added the tests as suggested.